gateway/ln-gateway/src/lib.rs

```diff
@@ -164,6 +171,29 @@ impl LnGateway {
         Ok(())
     }
 
+    async fn assign_channel_id(&self) -> u64 {
+        // The gateway deterministically assigns a channel id (u64) to each federation connected.
+        // At present, the maximum number of federations that can be referenced in the actors map
+        // is equivalent to `usize.max_value()`,
+
+        let mut channels = self.channels.lock().await;
+        let channels_n = channels.len() as u64;
+
+        let actors_n = self.actors.lock().await.len() as u64;
+
+        if actors_n == channels_n {
+            channels.insert(actors_n);
+            actors_n
+        } else {
+            let mut channel_id = 0;
+            while channels.contains(&channel_id) {
+                channel_id += 1;
+            }
+            channels.insert(channel_id);
+            channel_id
```

Comment:
Could this be implemented more easily using [`AtomicU64::fetch_add`](https://doc.rust-lang.org/std/sync/atomic/struct.AtomicU64.html#method.fetch_add)? This wouldn't allow recycling IDs, but it's unclear if that's necessary.

Reply:
Yes! Besides the potential limitation on recycling ids and a new overflow problem when using `fetch_add`, this turned out to be a superior approach.
- we start off at channel id `0 = AtomicU64::new(0)`. Might consider starting off at `1`, reserving `0` to solve a challenge with _wrap around on overflow_ behavior of `fetch_add`
- whenever we are connecting a new federation, we assign the current `u64` value as channel id to the new actor created, and pick a new channel id for next time. `let channel_id = self.channel_generator.fetch_add(1, Ordering::SeqCst)`
- on gateway reload, we read previously assigned channel ids, and start the AtomicU64 count from the largest id (last known assigned id). Ref **load_federation_actors()**